### PR TITLE
merge NCO's changes to github repository

### PR DESCRIPTION
**Pull request by ShunLiu-NOAA**
radarl2 package was moved from vlab to github. The changes for WCOSS2 implementation made by NCO need to be merged back to develop branch of this repository.
The changes were made for processing the small bufr files using bufr-tranjb.

### CODE REVIEW

File: ecf/dev/sub_level2.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#module load PrgEnv-intel/8.1.0
+#module load intel/19.1.3.304
+#module load craype
+#module load cray-mpich
+#module load grib_util/1.2.2
+#module load prod_util/2.0.5
+#module load prod_envir/2.0.4
+#module load cray-python
+
+source ~/.bashrc
```
Comment (aerorahul) [inline ecf/dev/sub_level2.sh]: what?

File: ecf/dev/del.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+
```
Comment (aerorahul) [inline ecf/dev/del.sh]: Does this script need to live in a public repository?
It is hardwired for a single user.

File: ecf/dev/jradar_level2.ecf [hunk 3]
```diff
@@ -9,8 +9,9 @@
 
 set -x
 
+export HOME=/lfs/h2/emc/da/noscrub/Shun.Liu/nwprod/radarl2.v1.2.2
```
Comment (aerorahul) [inline ecf/dev/jradar_level2.ecf]: Can we atleast make a little attempt to make our scripts generic?

File: ecf/dev/sub_level2.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash -login
```
Comment (aerorahul) [inline ecf/dev/sub_level2.sh]: why does one need a script to do:

cd directory_name
qsub script_name

File: jobs/JRADAR_LEVEL2 [hunk 5]
```diff
@@ -55,25 +45,14 @@ export DATAO=$DATA
...
-export outid="LL$job"
-export jobid="${outid}.o${pid}"
 export pgmout="OUTPUT.${pid}"
 
 
 ####################################
 # Specify Execution Areas
 ####################################
-# JY export HOMEradarl2=${HOMEradarl2:-${NWROOT}/radar.${radar_ver}}
+#export HOMEradarl2=${HOMEradarl2:-${PACKAGEROOT}/radarl2.${radarl2_ver}}
```
Comment (aerorahul) [inline jobs/JRADAR_LEVEL2]: Just delete the commented line.

File: jobs/JRADAR_LEVEL2 [hunk 6]
```diff
@@ -83,36 +62,31 @@ export HOMEbufr=${HOMEbufr:-${OPSROOT}/../prod/decoders/decod_shared}
...
-#export HOMEsatingest=${HOMEsatingest:-$NWROOTp3/obsproc_satingest.${obsproc_satingest_ver}}
+#NO NEED for satingest now ushing tranjb module 
+## JY export HOMEsatingest=${HOMEsatingest:-$NWROOTp3/obsproc_satingest.${obsproc_satingest_ver}}
+#export HOMEsatingest=${HOMEsatingest:-${PACKAGEROOT}/obsproc_satingest.${obsproc_satingest_ver}}
 #export USHsatingest=${HOMEsatingest}/ush
 #export EXECobsproc_satingest=${HOMEsatingest}/exec
 
+## JY added the following - 10/20
+# export TRANush=${USHsatingest}/bufr_tranjb.sh
+
```
Comment (aerorahul) [inline jobs/JRADAR_LEVEL2:72]: Seems irrelevant to radarl2 processing.
Please remove.

File: jobs/JRADAR_LEVEL2 [hunk 7]
```diff
@@ -130,7 +104,12 @@ postmsg "$msg"
 
 set -x
 #export NUMPROC=$LSB_DJOB_NUMPROC
+# JY check here
+echo "DEBUG: NCPUS=$NCPUS"
+
+# JY export NUMPROC=32
```
Comment (aerorahul) [inline jobs/JRADAR_LEVEL2]: echo statements for debugging don't need to be present in production

File: jobs/JRADAR_REFLECTIVITY_MOSAIC [hunk 8]
```diff
@@ -59,8 +54,8 @@ export hisdir=${hisdir:-${DCOMROOT}/../../${envir}/dcom/nexrad_level2}
 ####################################
 # set mosaic dump history file 
 ####################################
-# JY currenttime=`$NDATE ``date +%M`
-currenttime=${PDY}${cyc}`date +%M`
+currenttime=`$NDATE ``date +%M`
+#currenttime=${PDY}${cyc}`date +%M`
```
Comment (aerorahul) [inline jobs/JRADAR_REFLECTIVITY_MOSAIC]: remove.

File: jobs/JRADAR_REFLECTIVITY_REF2GRB [hunk 9]
```diff
@@ -36,36 +29,21 @@ export RUN=radar
...
-####################################
-export outid="LL$job"
-export jobid="${outid}.o${pid}"
 export pgmout="OUTPUT.${pid}"
 
 ####################################
 # Specify Execution Areas
 ####################################
-export HOMEradarl2=${HOMEradarl2:-${NWROOT}/radar_level2.${radar_ver}}
+#export HOMEradarl2=${HOMEradarl2:-${NWROOT}/radar_level2.${radar_ver}}
```
Comment (aerorahul) [inline jobs/JRADAR_REFLECTIVITY_REF2GRB]: delete

File: jobs/JRADAR_REFLECTIVITY_REF2GRB [hunk 10]
```diff
@@ -36,36 +29,21 @@ export RUN=radar
...
-##############################
-# Set up the UTILITIES
-##############################
-#export utilscript=/nwprod/util/ush
-#export utilexec=/nwprod/util/exec
 
-# JY mydate=`${NDATE}`
-mydate=${PDY}${cyc}
+mydate=`${NDATE}`
+#mydate=${PDY}${cyc}
```
Comment (aerorahul) [inline jobs/JRADAR_REFLECTIVITY_REF2GRB]: delete

File: jobs/JRADAR_REFLECTIVITY_REF2GRB [hunk 11]
```diff
@@ -104,11 +76,11 @@ mkdir -p $COMOUT
 $SCRIPTSradar/exradar_reflectivity_ref2grb.sh
 cat $pgmout
 
-# JY mydate=`$NDATE -48`
-mydate=`$NDATE -48 ${PDY}${cyc}`
+mydate=`$NDATE -48`
+#mydate=`$NDATE -48 ${PDY}${cyc}`
```
Comment (aerorahul) [inline jobs/JRADAR_REFLECTIVITY_REF2GRB]: delete

File: jobs/JRADAR_REFLECTIVITY_REF2GRB [hunk 12]
```diff
@@ -104,11 +76,11 @@ mkdir -p $COMOUT
...
 
-# JY COMINm1=${COMINm1:-${COMROOT}/${NET}/${RUN}.$mydate}
-COMINm1=${COMINm1:-$(compath.py ${envir}/${NET}/${radarl2_ver}/${RUN}.$mydate)}
+COMINm1=${COMINm1:-${COMROOT}/${NET}/${RUN}.$mydate}
+#COMINm1=${COMINm1:-$(compath.py ${envir}/${NET}/${radarl2_ver}/${RUN}.$mydate)}
```
Comment (aerorahul) [inline jobs/JRADAR_REFLECTIVITY_REF2GRB]: delete

File: sorc/build_wsr88d.sh [hunk 13]
```diff
@@ -21,4 +21,6 @@ make
 cd $pwd/wsr88d_level2.fd/; 
 make
 
-cp wsr88d_level2 ../../exec 
+mkdir -p -m 775 ../../exec
```
Comment (aerorahul) [inline sorc/build_wsr88d.sh:24]: why `-m 775`?

**Review by aerorahul — CHANGES_REQUESTED** [review]
See comments

File: ecf/dev/sub_level2.sh (hunk 1 above)
Comment (ShunLiu-NOAA) [reply ecf/dev/sub_level2.sh]: This is for development test script under ecf/dev. NCO will not use this. But I can remove this if you think it is not necessary.

File: ecf/dev/del.sh (hunk 2 above)
Comment (ShunLiu-NOAA) [reply ecf/dev/del.sh]: Good point. I will keep this in my branch for development test and remove it from "develop" branch

File: ecf/dev/sub_level2.sh (hunk 1 above)
Comment (aerorahul) [reply ecf/dev/sub_level2.sh]: yes. I don't think this script needs to live in a repository.
Instructions on how to use it should live in a README or something, which is what this script is essentially providing.

Comment (ShunLiu-NOAA) [reply ecf/dev/sub_level2.sh]: this script is deleted

File: ecf/dev/del.sh (hunk 2 above)
Comment (ShunLiu-NOAA) [reply ecf/dev/del.sh]: The script is deleted

File: ecf/dev/jradar_level2.ecf (hunk 3 above)
Comment (ShunLiu-NOAA) [reply ecf/dev/jradar_level2.ecf]: added a variable "radar_level2_ver" so that the test code directory is associated with the version number.

File: ecf/dev/sub_level2.sh (hunk 4 above)
Comment (ShunLiu-NOAA) [reply ecf/dev/sub_level2.sh]: the script is deleted

File: jobs/JRADAR_LEVEL2 (hunk 5 above)
Comment (ShunLiu-NOAA) [reply jobs/JRADAR_LEVEL2]: done

File: jobs/JRADAR_LEVEL2 (hunk 6 above)
Comment (ShunLiu-NOAA) [reply jobs/JRADAR_LEVEL2:72]: This is the comments from NCO. The above lines are still used on WCOSS. module of bufr_tranjb is replacing the directory paths on WCOSS2. However, NCO still want to know the difference between WCOSS and WCOSS2 during WCOSS2 transition so we can remove this in the next CDF when we fully transit to WCOSS2.

File: jobs/JRADAR_LEVEL2 (hunk 7 above)
Comment (ShunLiu-NOAA) [reply jobs/JRADAR_LEVEL2]: done

File: jobs/JRADAR_REFLECTIVITY_MOSAIC (hunk 8 above)
Comment (ShunLiu-NOAA) [reply jobs/JRADAR_REFLECTIVITY_MOSAIC]: done

File: jobs/JRADAR_REFLECTIVITY_REF2GRB (hunk 9 above)
Comment (ShunLiu-NOAA) [reply jobs/JRADAR_REFLECTIVITY_REF2GRB]: done

File: jobs/JRADAR_REFLECTIVITY_REF2GRB (hunk 10 above)
Comment (ShunLiu-NOAA) [reply jobs/JRADAR_REFLECTIVITY_REF2GRB]: done

File: jobs/JRADAR_REFLECTIVITY_REF2GRB (hunk 11 above)
Comment (ShunLiu-NOAA) [reply jobs/JRADAR_REFLECTIVITY_REF2GRB]: done

File: jobs/JRADAR_REFLECTIVITY_REF2GRB (hunk 12 above)
Comment (ShunLiu-NOAA) [reply jobs/JRADAR_REFLECTIVITY_REF2GRB]: done

File: sorc/build_wsr88d.sh (hunk 13 above)
Comment (ShunLiu-NOAA) [reply sorc/build_wsr88d.sh:24]: NCO made this change. I think they want to set permission for directory in operation.

File: jobs/JRADAR_LEVEL2 (hunk 6 above)
Comment (aerorahul) [reply jobs/JRADAR_LEVEL2:72]: That sounds fair enough.
I did not know the origins of these lines.

File: sorc/build_wsr88d.sh (hunk 13 above)
Comment (aerorahul) [reply sorc/build_wsr88d.sh:24]: so they want `group-writable` permissions to the `exec` directory? 
Seems odd to me.

**Review by aerorahul — APPROVED** [review]
I have a couple of additional comments, but none stopping this PR from merging.

Re. `ecf/dev/jradar_level2.ecf`  What makes this an `ecf` script?  It seems to be a regular bash script.

File: sorc/build_wsr88d.sh (hunk 13 above)
Comment (ShunLiu-NOAA) [reply sorc/build_wsr88d.sh:24]: NCO made similar changes in 
JRADAR_LEVEL2:mkdir -p -m 775 ${COMOUT_SRCtank}
JRADAR_LEVEL2:    chmod 775 ./poescript$proc
JRADAR_LEVEL2:chmod 775 poescript
JRADAR_REFLECTIVITY_MOSAIC:    chmod 775 ./poescript$proc
JRADAR_REFLECTIVITY_MOSAIC:chmod 775 poescript
I guess this allows that more than one developer at NCO is able to check radarl2 output with their individual account.
